toolkit/tools/scheduler/scheduler.go

```diff
@@ -277,6 +277,26 @@ func startWorkerPool(agent buildagents.BuildAgent, workers, buildAttempts, check
 	return
 }
 
+// debugStuckNode is a debugging function that will print out the stuck node and all nodes that are blocking it.
+func debugStuckNode(buildState *schedulerutils.GraphBuildState, pkgGraph *pkggraph.PkgGraph, stuckNode *pkggraph.PkgNode, indent int) {
+	isStuck := !buildState.IsNodeAvailable(stuckNode)
+	indentSpaces := ""
+	for i := 0; i < indent; i++ {
+		indentSpaces += "  "
+	}
+	if isStuck {
+		logger.Log.Debugf("DEBUG: %s**STUCK** (%s)", indentSpaces, stuckNode.FriendlyName())
```

Comment:
wondering if this should be a Warning message so it shows up easier in  the pipeline builds?

Reply:
Good call. I'm thinking we might even want to do an error message here, since this is something that's not supposed to happen and it breaks the build.

Reply:
I can see if I can enlighten `PrintBuildSummary()` to do something similar, but the code will likely be more complex than this. I don't think we know before then if this is a "normal" failure where we get everything we need in the summary, or an abnormal failure where the blocking node isn't showing up.